### PR TITLE
Add resetSharedInstance to Distribute

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add resetSharedInstance to Distribute in order to run functional tests on Xamarin.

## Related PRs or issues

[AB#75371](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/75371)

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 1]
```diff
@@ -53,6 +53,12 @@
 
 static NSString *const kMSUpdateTokenURLInvalidErrorDescFormat = @"Invalid update token URL:%@";
 
+/**
+ * Singleton.
+ */
+static MSDistribute *sharedInstance;
+static dispatch_once_t onceToken;
```
Comment (MatkovIvan) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]:
```suggestion

static dispatch_once_t onceToken;
```

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 2]
```diff
@@ -102,10 +108,10 @@ - (instancetype)init {
 #pragma mark - MSServiceInternal
 
 + (instancetype)sharedInstance {
-  static id sharedInstance = nil;
-  static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-    sharedInstance = [[MSDistribute alloc] init];
+    if (sharedInstance == nil) {
```
Comment (MatkovIvan) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: No need for `if` here

Comment (owlstronaut) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: We seem to have an `if` here in other modules. Why don't we need one here?

Comment (MatkovIvan) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Because of `dispatch_once` usage

Comment (clpolet) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: The dispatch once will make sure there's only one assignment across calls and threads. It was probably overlooked in other places.

**Pull request merged**